tribe-ext-calendar-widget-areas.php

```diff
@@ -0,0 +1,472 @@
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU General Public License for more details.
+ */
+
+namespace Tribe\Extensions\Calendar_Widget_Areas;
+
+use Tribe__Autoloader;
+use Tribe__Events__Main;
+use Tribe__Extension;
+
+/**
+ * Define Constants
+ */
+
+if ( ! defined( __NAMESPACE__ . '\NS' ) ) {
+	define( __NAMESPACE__ . '\NS', __NAMESPACE__ . '\\' );
+}
+
+if ( ! defined( NS . 'PLUGIN_TEXT_DOMAIN' ) ) {
+	// `Tribe\Extensions\Calendar_Widget_Areas\PLUGIN_TEXT_DOMAIN` is defined
+	define( NS . 'PLUGIN_TEXT_DOMAIN', 'tribe-ext-calendar-widget-areas' );
+}
+
+// Do not load unless Tribe Common is fully loaded and our class does not yet exist.
+if (
+	class_exists( 'Tribe__Extension' )
+	&& ! class_exists( Main::class )
+) {
+	/**
+	 * Extension main class, class begins loading on init() function.
+	 */
+	class Main extends Tribe__Extension {
+
+		/**
+		 * @var Tribe__Autoloader
+		 */
+		private $class_loader;
+
+		/**
+		 * Setup the Extension's properties.
+		 *
+		 * This always executes even if the required plugins are not present.
+		 */
+		public function construct() {
+			$this->add_required_plugin( 'Tribe__Events__Main' );
+		}
+
+		/**
+		 * Get this plugin's options prefix.
+		 *
+		 * Settings_Helper will append a trailing underscore before each option.
+		 *
+		 * @see \Tribe\Extensions\Calendar_Widget_Areas\Settings::set_options_prefix()
+		 *
+		 * @return string
+		 */
+		private function get_options_prefix() {
+			return (string) str_replace( '-', '_', PLUGIN_TEXT_DOMAIN );
+		}
+
+		/**
+		 * Get Settings instance.
+		 *
+		 * @return Settings
+		 */
+		private function get_settings() {
+			if ( empty( $this->settings ) ) {
+				$this->settings = new Settings( $this->get_options_prefix() );
+			}
+
+			return $this->settings;
+		}
+
+		/**
+		 * Extension initialization and hooks.
+		 */
+		public function init() {
+			load_plugin_textdomain( PLUGIN_TEXT_DOMAIN, false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+			if ( ! $this->php_version_check() ) {
+				return;
+			}
+
+			$this->class_loader();
+
+			$this->get_settings();
+
+			// Register all widget areas.
+			add_action( 'widgets_init', [ $this, 'register_sidebars' ] );
+
+			add_action( 'init', [ $this, 'enqueue_styles' ] );
+
+			// Add filters and actions based on choices.
+			foreach ( $this->get_enabled_areas_full_details() as $value ) {
+				if ( method_exists( $this, $value['method'] ) ) {
+					if ( empty( $value['priority'] ) ) {
+						$priority = 10;
+					} else {
+						$priority = $value['priority'];
+					}
+
+					$priority = apply_filters( 'tribe_ext_calendar_widget_area_priority', $priority, $value );
+
+					$priority = (int) $priority;
+
+					if ( ! empty( $value['filter'] ) ) {
+						add_filter( $value['hook'], [ $this, $value['method'] ], $priority );
+					} else {
+						add_action( $value['hook'], [ $this, $value['method'] ], $priority );
+					}
```

Comment:
To improve readability I think changing the body of this one and removing the `!` operator will help.

```suggestion
					if ( empty( $value['filter'] ) ) {
						add_action( $value['hook'], [ $this, $value['method'] ], $priority );					
					} else {
						add_filter( $value['hook'], [ $this, $value['method'] ], $priority );
					}
```

Reply:
I like this suggestion.